perceval/runtime/job_group.py

```diff
@@ -0,0 +1,281 @@
+                                             FileFormat.TEXT))
+        return group_data
+
+    def _write_job_group_to_disk(self):
+        # writes job group data to disk
+        jgrp_dir = JobGroup._get_job_group_dir()
+        PersistentData().write_file(os.path.join(jgrp_dir, self._file_name),
+                                    json.dumps(self._group_info), FileFormat.TEXT)
+
+    def _load_job_group(self):
+        # Creates a Job Group by loading an existing one from file
+        group_data = self._read_job_group_from_disk()
+
+        self._group_info['created_date'] = group_data['created_date']
+        self._group_info['modified_date'] = JobGroup._get_current_datetime()
+        self._group_info['job_group_data'] = group_data['job_group_data']
+        # Write to disk (necessary to update modification time)
+        self._write_job_group_to_disk()
+
+    @staticmethod
+    def list_saved_job_groups():
+        """
+        Returns a list of filenames of all JobGroups saved to disk
+        """
+        jgrp_path = JobGroup._get_job_group_dir()
+        files = [f for f in os.listdir(jgrp_path) if f.endswith(FILE_EXT_JGRP)]
+        return files
+
+    @staticmethod
+    def _job_group_exists(name: str):
+        # returns True if a JobGroup with an identical name is already saved on disk
+        saved_job_groups = JobGroup.list_saved_job_groups()
+        return any([filename.split('.')[0] == name for filename in saved_job_groups])
+
+    def add(self, job_to_add: RemoteJob, **kwargs):
+        """
+        Adds information of the new RemoteJob to an existing Group.
+        Saves the data in a chronological order in the group (each entry is
+        a dictionary of necessary information - status, id, body, metadata)
+        """
+        if not isinstance(job_to_add, RemoteJob):
+            raise TypeError('Only a RemoteJob can be added to the group')
+
+        job_info = dict()
+        job_info['id'] = job_to_add.id
+
+        if job_to_add.id is None:
+            # set status to None for Jobs not sent to cloud
+            job_info['status'] = None
+            # Save information inside body (circuit, payload, etc.) to send job later
+            job_info['body'] = job_to_add._request_data
+            max_samples = kwargs.get('max_samples', MAX_SAMPLE_DEF_GROUP)
+            job_info['args'] = {"max_samples": max_samples}
+        else:
+            job_info['status'] = job_to_add.status()
+
+        if not job_info['status'] == 'SUCCESS':
+            # save metadata for any job that did not run successfully or was not sent
+            job_info['metadata'] = {'headers': job_to_add._rpc_handler.headers,
+                                    'platform': job_to_add._rpc_handler.name,
+                                    'url': job_to_add._rpc_handler.url}
+
+        self._group_info['job_group_data'].append(job_info)  # save changes in object
```

Comment:
Do you have a system refusing job id duplicates?

Reply:
I did not, implemented.